google-datacatalog-qlik-connector/tests/google/datacatalog_connectors/qlik/scrape/engine_api_helper_test.py

```diff
@@ -0,0 +1,116 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import unittest
+from unittest import mock
+
+from google.datacatalog_connectors.qlik.scrape import \
+    constants, engine_api_helper
+
+from . import scrape_ops_mocks
+
+_SCRAPE_PACKAGE = 'google.datacatalog_connectors.qlik.scrape'
+
+
+@mock.patch(f'{_SCRAPE_PACKAGE}.engine_api_helper.websockets.connect',
+            new_callable=scrape_ops_mocks.AsyncContextManager)
+class EngineAPIHelperTest(unittest.TestCase):
+
+    def setUp(self):
+        # Set a constant seed to ensure generated numbers will always be the
+        # same when running the tests.
+        random.seed(1)
+
+        self.__helper = engine_api_helper.EngineAPIHelper(
+            server_address='https://test-server')
+
+    def test_constructor_should_set_instance_attributes(self, mock_websocket):
+        attrs = self.__helper.__dict__
+
+        self.assertEqual('https://test-server',
+                         attrs['_EngineAPIHelper__server_address'])
+        self.assertEqual('wss://test-server',
+                         attrs['_EngineAPIHelper__base_api_endpoint'])
+        self.assertIsNotNone(attrs['_EngineAPIHelper__common_headers'])
+
+    def test_websocket_connection_should_use_cookie(self, mock_websocket):
+        mock_websocket.return_value.__enter__.return_value.set_data([
+            {
+                'id': 2202,
+                'result': {
+                    'qReturn': {}
+                }
+            },
+        ])
+
+        fake_cookie = scrape_ops_mocks.FakeQPSSessionCookie()
+        self.__helper.get_sheets('app-id', fake_cookie)
+
+        mock_websocket.assert_called_once()
+
+        extra_headers = self.__helper.__dict__[
+            '_EngineAPIHelper__common_headers'].copy()
+        extra_headers['Cookie'] = 'X-Qlik-Session=Test cookie'
+        mock_websocket.assert_called_with(
+            uri=f'wss://test-server/app/app-id?Xrfkey={constants.XRFKEY}',
+            extra_headers=extra_headers)
+
+    def test_get_sheets_should_return_list_on_success(self, mock_websocket):
+        mock_websocket.return_value.__enter__.return_value.set_data([
+            {
+                'id': 2202,
+                'result': {
+                    'qReturn': {
+                        'qHandle': 1
+                    }
+                }
+            },
+            {
+                'id': 9326,
+                'result': {
+                    'qList': []
+                }
+            },
+        ])
+
+        sheets = self.__helper.get_sheets({'id': 'app-id'}, mock.MagicMock())
+
+        self.assertIsNotNone(sheets)
```

Comment:
The test case: `test_get_sheets_should_return_list_on_success`

The assert should ensure that the method returns a list from the name **return_list_on_success**.

Also if you think it makes sense improving this test case, you can test the other scenario on a second test case where if the response does not contain the generated `request_id`, it will return None.

Reply:
Makes sense, thanks! I've improved the assertions and added the proposed test case.